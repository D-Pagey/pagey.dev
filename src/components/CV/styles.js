import styled from 'styled-components';

export const Wrapper = styled.div`
  /* border: 1px solid blue; */
  margin: 0 auto;
  max-width: 750px;
`;

export const List = styled.ul`
  list-style: none;
  padding: 0;
`;

export const Item = styled.li`
  color: blue;
`;

export const CompanyLink = styled.a`
  cursor: pointer;
`;
