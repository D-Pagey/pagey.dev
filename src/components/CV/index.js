import React from 'react';
import cytora from './assets/cytora.svg';
import * as S from './styles';

const CV = () => (
  <S.Wrapper>
    <h1>Curriculum Vitae</h1>
    <h2>Dan Page</h2>

    <p>
      I am a Software Developer that loves building applications in JavaScript. I’m always thinking about coding,
      whether that is what I want to learn next or what I’m going to build next.{' '}
    </p>

    <h2>Where I have worked:</h2>
    <S.List>
      <S.Item>
        <S.CompanyLink href="https://cytora.com/" target="_blank">
          <img src={cytora} alt="cytora" />
        </S.CompanyLink>
        <p>
          I joined as the first Front End Developer to build out UIs from scratch on top of the API platform. I&apos;m
          also creating a reusable component library to be able to build future UIs quickly and consistently.
        </p>
      </S.Item>
      <S.Item>
        <S.CompanyLink href="https://www.joindrover.com/" target="_blank">
          Drover
        </S.CompanyLink>
        <p>
          I joined Drover as a Junior Front End Developer and my first project was working solo on rebuilding Ruby on
          Rails pages into React. Whilst I was confident working in React, I quickly learnt how to work in a large
          codebase with lots of shared components (and minimal tests!).
        </p>

        <p>
          After 6 months I have been promoted to a mid-level Developer and I believe that I have shown myself to be an
          equal part of the Front End team. I am constantly looking for ways to improve our codebase and workflow. I
          have taken advantage of working in a small team by suggesting and implementing improvements quickly.
          Currently, I am the flag bearer for doing more unit testing and I spend a lot of my free time improving myself
          in this area.
        </p>
      </S.Item>
    </S.List>
  </S.Wrapper>
);

export default CV;
