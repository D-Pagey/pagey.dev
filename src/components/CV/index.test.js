import React from 'react';
import CV from '.';

describe('CV component', () => {
  it('should render', () => {
    const { container } = render(<CV />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
