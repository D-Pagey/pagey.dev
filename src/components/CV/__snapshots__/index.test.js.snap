// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CV component should render 1`] = `
.c0 {
  margin: 0 auto;
  max-width: 600px;
}

.c1 {
  text-align: center;
  font-family: 'Merriweather';
}

.c2 {
  font-family: 'Merriweather';
}

.c3 {
  list-style: none;
  padding: 0;
}

.c4 {
  margin: 2rem 0 0;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1rem 0 0;
}

.c5 {
  cursor: pointer;
}

.c6 {
  display: block;
  margin: 0 auto;
  height: 50px;
  width: 150px;
}

.c8 {
  font-weight: 600;
  font-size: 1.125rem;
  margin: 0;
}

.c9 {
  font-style: italic;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
  padding: 0;
}

.c11 {
  background-color: skyblue;
  border-radius: 5px;
  font-size: 12px;
  padding: 0.25rem 0.5rem;
  margin: 0 0 0.5rem 0.5rem;
  text-transform: uppercase;
}

.c12 {
  background-color: pink;
  border-radius: 5px;
  font-size: 12px;
  padding: 0.25rem 0.5rem;
  margin: 0 0 0.5rem 0.5rem;
  text-transform: uppercase;
}

.c13 {
  background-color: lightgreen;
  border-radius: 5px;
  font-size: 12px;
  padding: 0.25rem 0.5rem;
  margin: 0 0 0.5rem 0.5rem;
  text-transform: uppercase;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Mini CV
  </h1>
  <h2
    class="c2"
  >
    Dan Page
  </h2>
  <p>
    I am a Software Developer that loves building applications in JavaScript. I’m always thinking about coding, whether that is what I want to learn next or what I’m going to build next.
     
  </p>
  <ul
    class="c3"
  >
    <li
      class="c4"
    >
      <a
        class="c5"
        href="https://cytora.com/"
        target="_blank"
      >
        <img
          alt="cytora"
          class="c6"
          src="cytora.svg"
        />
      </a>
      <div
        class="c7"
      >
        <h5
          class="c8"
        >
          Front End Engineer
        </h5>
        <span
          class="c9"
        >
          May '19 - Present
        </span>
      </div>
      <p>
        I joined as the first Front End Developer to build out UIs from scratch on top of the API platform. I'm also creating a reusable component library to enable multiple teams to build UIs quickly and consistently.
      </p>
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          react
        </li>
        <li
          class="c11"
        >
          react-context
        </li>
        <li
          class="c11"
        >
          styled components
        </li>
        <li
          class="c11"
        >
          storybook
        </li>
        <li
          class="c11"
        >
          jest
        </li>
        <li
          class="c11"
        >
          react-testing-library
        </li>
        <li
          class="c11"
        >
          cypress
        </li>
      </ul>
    </li>
    <li
      class="c4"
    >
      <a
        class="c5"
        href="https://www.joindrover.com/"
        target="_blank"
      >
        <img
          alt="drover"
          class="c6"
          src="drover.svg"
        />
      </a>
      <div
        class="c7"
      >
        <h5
          class="c8"
        >
          Front End Developer
        </h5>
        <span
          class="c9"
        >
          Sept '18 - May '19
        </span>
      </div>
      <p>
        I joined Drover as a Junior Front End Developer and my first project was working on rebuilding Ruby on Rails pages in React. The main learning curve came from using shared components that had no unit or integration tests, which made refactoring challenging.
      </p>
      <p>
        After 6 months I was promoted to a mid-level Developer and put on a 3 month greenfield project to rebuild the entire user booking flow. I was the flag bearer for doing more unit and integration testing and so I spent a lot of my free time learning about jest, enzyme, react-testing-library and cypress.
      </p>
      <ul
        class="c10"
      >
        <li
          class="c12"
        >
          react
        </li>
        <li
          class="c12"
        >
          redux
        </li>
        <li
          class="c12"
        >
          styled components
        </li>
        <li
          class="c12"
        >
          storybook
        </li>
        <li
          class="c12"
        >
          jest
        </li>
        <li
          class="c12"
        >
          enzyme
        </li>
        <li
          class="c12"
        >
          cypress
        </li>
      </ul>
    </li>
    <li
      class="c4"
    >
      <a
        class="c5"
        href="https://clearscore.com/"
        target="_blank"
      >
        <img
          alt="clearscore"
          class="c6"
          src="clearscore.svg"
        />
      </a>
      <div
        class="c7"
      >
        <h5
          class="c8"
        >
          Talent Acquisition
        </h5>
        <span
          class="c9"
        >
          May '17 - Sept '18
        </span>
      </div>
      <p>
        I joined ClearScore as roughly employee #100 and, by the time I left, the company had grown to around 200 employees internationally.
      </p>
      <p>
        Beyond my day-to-day recruitment role, I created and ran a company-wide interview training programme that focused on bringing all interviewers on to the same page. This aimed to ensure the interview process was fair and unbiased, provided candidates with useful feedback and raised awareness around unconscious bias.
      </p>
      <p>
        I also made multiple contributions to the front-end production code base which varied from updating integrations on the careers page to bug fixes on the home page.
      </p>
      <ul
        class="c10"
      >
        <li
          class="c13"
        >
          es6
        </li>
        <li
          class="c13"
        >
          handlebars
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
